### PR TITLE
Reinstate S3 repo properties to maintain BWC

### PR DESCRIPTION
**Pull request by codebrain**

### CODE REVIEW

File: src/Nest/Modules/SnapshotAndRestore/Repositories/S3Repository.cs [hunk 1]
```diff
@@ -125,6 +147,20 @@ public class S3RepositorySettingsDescriptor
 
 		public S3RepositorySettingsDescriptor(string bucket) => Self.Bucket = bucket;
 
+#pragma warning disable 618
```
Comment (russcam) [inline src/Nest/Modules/SnapshotAndRestore/Repositories/S3Repository.cs]: Can remove pragma directive I think, now that the methods are marked `[Obsolete]`

File: src/Nest/Modules/SnapshotAndRestore/Repositories/S3Repository.cs [hunk 2]
```diff
@@ -7,14 +7,26 @@ public interface IS3Repository : IRepository<IS3RepositorySettings> { }
...
-		public S3Repository(IS3RepositorySettings settings) => Settings = settings;
+		public S3Repository(S3RepositorySettings settings) => Settings = settings;
 
 		public IS3RepositorySettings Settings { get; set; }
 		public string Type { get; } = "s3";
 	}
 
 	public interface IS3RepositorySettings : IRepositorySettings
 	{
+		[Obsolete("Removed in Elasticsearch 6.0, please specify secure settings in the keystore. See https://www.elastic.co/guide/en/elasticsearch/reference/6.x/secure-settings.html")]
```
Comment (russcam) [inline src/Nest/Modules/SnapshotAndRestore/Repositories/S3Repository.cs]: slight nitpick: Not overly keen on links in the attributes as they can drift. Maybe just
```
Removed in Elasticsearch 6.0. Please specify secure settings in the Elasticsearch keystore.
```
?

File: src/Nest/Modules/SnapshotAndRestore/Repositories/S3Repository.cs (hunk 1 above)
Comment (codebrain) [reply src/Nest/Modules/SnapshotAndRestore/Repositories/S3Repository.cs]: :+1:

File: src/Nest/Modules/SnapshotAndRestore/Repositories/S3Repository.cs (hunk 2 above)
Comment (codebrain) [reply src/Nest/Modules/SnapshotAndRestore/Repositories/S3Repository.cs]: :+1:

**Pull request merged**